### PR TITLE
448 add entities queries to rust backend

### LINKED ISSUE #448: Add entities queries to Rust Backend
Currently rust backend doesn't support entities query type.
image

### PR DESCRIPTION
**Pull request by jmbrunskill**
Fixes #448 

## Description
This creates a /v1/graphql api that's mostly compatible with the old universal codes one.

Updates dgraph schema to add Category ('Drug', 'Consumable' or 'Other') for every Entity
Update dgraph schema to add `type` as this is not queryable via graphql if it's only represented by the dgraph type...

### Tests:

Some tests cases have been added as `jest` tests but we probably need to increase our rust based test cases.

Would also be good to compare results with different queries, or more importantly make sure what you expect to have returned from your query actually is!

### CODE REVIEW

File: backend/dgraph/src/entities.rs [hunk 1]
```diff
@@ -41,17 +80,117 @@ query EntitiesQuery($search: String = "", $first: Int = 10, $offset: Int = 0) {
...
-            first: 10,
-            offset: 0,
+            filter: DgraphFilter {
+                description: Some(DgraphFilterType {
+                    regexp: Some("/Heparin Sodium.*/".to_string()),
+                    ..Default::default()
+                }),
+                category: Some(DgraphFilterType {
+                    r#in: Some(vec!["Drug".to_string()]),
+                    ..Default::default()
```
Comment (jmbrunskill) [inline backend/dgraph/src/entities.rs:98]: Some helper functions like we have for our filters e.g. 
`let permission_filter = UserPermissionFilter::new().user_id(EqualFilter::equal_to(&ctx.user_id));`

I think we can wait to see if that's helpful later on? Or I could add to this PR?

Comment (lache-melvin) [reply backend/dgraph/src/entities.rs:98]: Ah yeah, those filters are nice, but can be added later for sure :)

**Comment by lache-melvin** [comment]
Oh the other thing would be that it would be good to get those integration tests moved over into Rust, but if you'd like to get this in and create another issue for that I think that's all good - but will be great to get rid of the data-service folder when we can!

**Comment by jmbrunskill** [comment]
> t would be good to get those integration tests moved over into Rust,

https://github.com/msupply-foundation/unified-codes/issues/458

**Comment by jmbrunskill** [comment]
> I see you haven't made the entity queries available at `/graphql` yet... I know there's a bit of extra work in it, in terms of updating the filters etc... I think this should be the next cab off the ranks as we do need these queries exposed there before we can merge any UI PRs

Created a new issue for this https://github.com/msupply-foundation/unified-codes/issues/459

Sorry for slow progress ended up mostly in meetings, and didn't get much time today!

File: backend/graphql_v1/universal_codes/src/types/entity_sort.rs
[the remark this answers is not in the document]
Comment (jmbrunskill) [reply backend/graphql_v1/universal_codes/src/types/entity_sort.rs:1]: Ah yes, not sure it needed but does at least align the names :)

[the remark this answers is not in the document]
Comment (jmbrunskill) [reply backend/graphql_v1/universal_codes/src/types/entity_sort.rs:1]: [e8f1633](https://github.com/msupply-foundation/unified-codes/pull/456/commits/e8f1633ef49742f066aba5353f83a999410455a2)

**Comment by lache-melvin** [comment]
> Sorry for slow progress ended up mostly in meetings, and didn't get much time today!

No worries! It's great to have all this work in James 🤩

**Pull request merged**